WordPress/Classes/Models/AbstractPost.swift

```diff
@@ -95,4 +95,23 @@ extension AbstractPost {
             return nil
         }
     }
+
+    // MARK: - Updating the Remote Status
+    /// Updates the post after an upload failure.
+    ///
+    /// - Important: This logic could have been placed in the setter for `remoteStatus`, but it's my belief
+    ///     that our code will be much more resilient if we decouple the act of setting the `remoteStatus` value
+    ///     and the logic behind processing an upload failure.  In fact I think the `remoteStatus` setter should
+    ///     eventually be made private.
+    ///
+    @objc
+    func failedToUpload() {
```

Comment:
⚠️  Not sure I'd agree that this is in `AbstractPost`. I believe that model classes should be lightweight. No business logic as much as possible. I say “as much as possible” because sometimes short methods makes our lives easier. However, I'm not confident about this one because it updates `remoteStatus` which is a critical piece of the uploading system. 

@diegoreymendez and I had a discussion related to this today. He's going to make a similar method for `Media`. Personally, I'd rather have a `UseCase/Interactor` class for this but I wouldn't recommend it at this point. I'm not sure if `PostService` is the best location for this but I think it's a good start. 

We should also update the failure block in `PostService` to use the same routine. 

https://github.com/wordpress-mobile/WordPress-iOS/blob/9d67d09955aa5aa2243fdffc50b7490e23b3041e/WordPress/Classes/Services/PostService.m#L257-L272

What do you think?

Reply:
I completely agree @shiki. Considering your feedback, I'm probably going to implement this as part of an intermediate PR.

Good feedback.

Reply:
PS: just realized this is NOT my PR!  @leandroalonso I can show you the plan I had for this and we can work on it tomorrow if you want!

Reply:
Moved to `PostService`.

Let me know what you guys think.